transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/SelectTransferMethodPresenter.java

```diff
@@ -150,11 +150,27 @@ public void onKeysLoaded(@Nullable final HyperwalletTransferMethodConfigurationK
                                         new ArrayList<>(key.getCurrencies(countryCode)) :
                                         new ArrayList<Currency>();
 
+                                Country selectedCountry = null;
+                                for (Country country : key.getCountries()) {
+                                    if (country.getCode().equals(countryCode)) {
+                                        selectedCountry = country;
+                                        break;
+                                    }
+                                }
+                                // Attempt to get the default currency code for the selected country
+                                String selectedCurrencyCode = getDefaultCurrencyCode(selectedCountry, key, countryCode);
+
```

Comment:
Could you change `List<Currency> currencies` to `Set<Currency> currencyCodes` and move this check statement inside the  `getDefaultCurrencyCode` method ?